webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala

```diff
@@ -530,18 +532,47 @@ class SystemVariableServiceImpl(
     val varNodeGroups = systemVariableSpecService.get("RUDDER_NODE_GROUPS_VARS").toVariable(Seq(stringNodeGroupsVars))
     val varNodeGroupsClasses = systemVariableSpecService.get("RUDDER_NODE_GROUPS_CLASSES").toVariable(Seq(stringNodeGroupsClasses))
 
-    // If Syslog is disabled, we force HTTPS
-    val varNodeReportingProtocol = getSyslogProtocolDisabled() match {
-      case Full(true)  => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(AgentReportingHTTPS.value))
-      case Full(false) => nodeInfo.nodeReportingConfiguration.agentReportingProtocol match {
-                            case Some(protocol) => systemVariableSpecService.get("REPORTING_PROTOCOL").toVariable(Seq(protocol.value))
-                            case _              => getProp("REPORTING_PROTOCOL", () =>  getReportProtocolDefault().map(_.value))
-                          }
-      case eb:EmptyBox =>
-        val e = eb ?~! s"Failed to get information on syslog protocol global status, fallbacking to default value."
-        logger.error( e.messageChain )
-        getProp("REPORTING_PROTOCOL", () => getReportProtocolDefault().map(_.value))
-    }
+    // If Syslog is disabled, we force HTTPS.
+    val varNodeReportingProtocol = {
+      // By default, we are tolerant: it's far worse to fail a generation while support is ok than to
+      // succeed and have a node that doesn't rerport.
+      // => version must exists and and starts by 2.x, 3.x, 4.x, 5.x
+      def versionHasHttps(maybeVersion: Option[AgentVersion]): Boolean = {
+        maybeVersion match {
+          case None    =>
+            true // we don't know, so say HTTPS is supported
+          case Some(v) =>
+            val forbiden = List("2.", "3.", "4.", "5.")
+            !forbiden.exists(x => v.value.startsWith(x))
+        }
+      }
+
+      getSyslogProtocolDisabled().flatMap {
+        case true  => Full(AgentReportingHTTPS)
+        case false => nodeInfo.nodeReportingConfiguration.agentReportingProtocol match {
+          case Some(protocol) => Full(protocol)
+          case _              => getReportProtocolDefault()
+        }
+      }.flatMap{ reportingProtocol => // If HTTPS is used on a node that does support it, we fails.
+        // as we don't have capalities and version is not reliable, we only fail when we are sure:
+        // - version provided and starts by 2.x, 3.x, 4.x, 5.x
+        // - agent is DSC
+        // Also, special case root, because not having root cause strange things.
+        val badVersion = nodeInfo.agentsName.find { agent => !versionHasHttps(agent.version) }
+        val dsc = nodeInfo.agentsName.find { agent => agent.agentType == AgentType.Dsc }
+        if(nodeInfo.id != Constants.ROOT_POLICY_SERVER_ID && (badVersion.nonEmpty || dsc.nonEmpty)) {
```

Comment:
shouldn't it fail only if HTTPS ?

Reply:
Hm. There was a check `reportingProtocol == AgentReportingHTTPS` but I beleive it was erased when I put appart version/dsc for error reporting.

Reply:
Never, ever change something after testing, even something that seems trivial :)